src/Membrain/Memory.hs

```diff
@@ -45,9 +53,14 @@ toBits :: Memory mem -> Natural
 toBits = coerce
 {-# INLINE toBits #-}
 
+
+nat :: forall (mem :: Nat) . KnownNat mem => Natural
+nat = natVal (Proxy @mem)
```

Comment:
Maybe lets put this function at the end of the module? So it appears after its every usage. And is not mixed with the exposed interface.